gcp/modules/gke/main.tf

```diff
@@ -0,0 +1,70 @@
+# copyright 2020 Datastax LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+resource "google_container_cluster" "container_cluster" {
+  name        = var.name
+  project     = var.project_id
+  description = "Demo GKE Cluster"
+  location    = var.location
+
+  remove_default_node_pool = true
+  initial_node_count       = var.initial_node_count
+
+  network    = var.network_link
+  subnetwork = var.subnetwork_link
+
+  master_auth {
+    # Setting an empty username and password explicitly disables basic auth
+    username = ""
+    password = ""
+
+    # Whether client certificate authorization is enabled for this cluster.
+    client_certificate_config {
+      issue_client_certificate = false
+    }
+  }
+
+  provisioner "local-exec" {
+    command = "sleep 120"
```

Comment:
If you are doing this to wait for the cluster to come up, this will not work.  We need to run a script to check if the cluster is up.